test/toDoListController.spec.js

```diff
@@ -0,0 +1,63 @@
+describe('MainController', function () {
+  beforeEach(module('ToDoList'));
+
+  var ctrl, scope;
+  beforeEach(inject(function($controller, $rootScope) {
+    scope = $rootScope.$new();
+    ctrl = $controller('MainController', {
+      $scope: scope
+    });
+  }));
+
+  it('initialises with no input', function () {
+    expect(scope.taskInput).toBeUndefined();
+  });
+
+  it('initialises with no tasks', function () {
+    expect(scope.tasks).toEqual([]);
+  });
+
+  var tasks = [{'name': 'Do my weekend challenege', 'done': false},
+              {'name': 'Actually understand Angular', 'done': false},
+              {'name': 'Finish Boris bike', 'done': false}];
+
+  describe('tasks', function () {
+    beforeEach(function () {
+      scope.taskInput = 'Do my weekend challenege';
+      scope.addTask();
+      scope.taskInput = 'Actually understand Angular';
+      scope.addTask();
+      scope.taskInput = 'Finish Boris bike';
+      scope.addTask();
+    });
+
+    it('stores tasks', function () {
+      expect(scope.tasks).toEqual(tasks);
+    });
+
+    it('does not allow blank tasks', function () {
+      scope.taskInput = '';
+      scope.addTask();
+      expect(scope.tasks).toEqual(tasks);
+    });
+
+    it('deletes tasks', function () {
+      // Karma doesn't seemd to know indexOf
+      // scope.deleteTask(tasks[0]);
```

Comment:
It would be good to get to the bottom of why this isn't working, as currently your .deleteTask method is sort of untested